### PR TITLE
[zosmf] Added a method to list the systems defined to z/OSMF

### PR DESCRIPTION
**Pull request by aadityasinha-dotcom**
Added a method to `list the systems` defined to z/OSMF
Signed-off-by: Aaditya Sinha [aadityasinha009@gmail.com](mailto:aadityasinha009@gmail.com)

### CODE REVIEW

**Review by t1m0thyj — CHANGES_REQUESTED** [review]

File: src/zosmf/zowe/zosmf_for_zowe_sdk/zosmf.py [hunk 1]
```diff
@@ -45,3 +45,17 @@ def get_info(self):
...
+        Returns
+        -------
+        json
+            A JSON containing the z/OSMF Info REST API data
+        """
+
+        custom_args = self.create_custom_request_arguments()
+        custom_args["url"] = "{}/systems".format(self.request_endpoint)
+        response_json = self.request_handler.perform_request("GET", custom_args, expected_code = [200])
+        return response_json
```
[the remark this answers is not in the document]
Comment (aadityasinha-dotcom) [reply src/zosmf/zowe/zosmf_for_zowe_sdk/zosmf.py:61]: I tried this approach but it decreases coverage by `0.13%`.

**Comment by aadityasinha-dotcom** [comment]
What should we do for `codecov/project`?

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
One more request - could you please add an integration (system) test to `tests/integration/test_zosmf.py`? It can be modelled after the existing integration test for `get_zosmf`, something like the following:

```
    def test_list_systems_should_return_valid_response(self):
        """Executing the list_systems method should return a valid response."""
        command_output = self.zosmf.list_systems()
        self.assertIsInstance(command_output, dict)
```

Unless you have access to a z/OS system, you won't be able to run the integration tests on your local machine, but that's fine, I can run the test for you to verify that it works.

File: tests/integration/test_zosmf.py [hunk 2]
```diff
@@ -17,3 +17,8 @@ def test_get_info_should_return_valid_response(self):
         """Executing the get_info method should return a valid response."""
         command_output = self.zosmf.get_info()
         self.assertIsInstance(command_output, dict)
+
+    def test_list_systems_should_return_valid_response(self):
```
Comment (aadityasinha-dotcom) [inline tests/integration/test_zosmf.py:21]: @t1m0thyj I have added the `integerated test` can you test that?

Comment (t1m0thyj) [reply tests/integration/test_zosmf.py:21]: Thanks for adding, it passes 👍

**Comment by t1m0thyj** [comment]
@aadityasinha-dotcom Please fix the failing DCO check, other than that everything looks good to me 🙂

**Comment by aadityasinha-dotcom** [comment]
@t1m0thyj Closing this PR as I have created a new PR #97

**Pull request closed without merging**